### DISCUSSION TITLE
Build and export to use with template engine replaces template strings with their raw values after loading

### DISCUSSION THREAD

**Discussion by eyenalxai** [vercel/next.js#28088] [Help]
I'm having exact same issue as described here, although I'm using Velocity template engine: 
https://stackoverflow.com/questions/62126466/next-js-build-and-export-for-template-engine

I have this component:
```
function Home() {
    return (
        <div>
            <h1>{"$commit.message"}</h1>
            <p>{"$commit.id"}</p>
        </div>
    )
}
```
https://github.com/yatxone/random-commit/blob/main/frontend/pages/index.tsx

Which I build using `next build && next export` command to later serve with my server app in Kotlin:
`var response = VelocityContent("static/index.html", mapOf("commit" to commit))`

Basically it takes an object and maps properties accordingly, _id_ property to `$commit.id` and _message_ to `$commit.message`.

But after loading the page it gets reverted back after a few milliseconds, as demonstrated below 😕

Is there something to be done about this? 🤔

https://user-images.githubusercontent.com/40147710/129404722-75967c5b-cdd9-42c6-8747-b9184722dfd5.mp4
